staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache_test.go

```diff
@@ -380,12 +399,23 @@ func TestWaitUntilFreshAndList(t *testing.T) {
 			return []string{pod.Spec.NodeName}, nil
 		},
 	})
-
+	fc := store.clock.(*testingclock.FakeClock)
 	// In background, update the store.
 	go func() {
+		for !fc.HasWaiters() {
+			time.Sleep(time.Millisecond)
+		}
 		store.Add(makeTestPodDetails("pod1", 2, "node1", map[string]string{"label": "value1"}))
 		store.Add(makeTestPodDetails("pod2", 3, "node1", map[string]string{"label": "value1"}))
 		store.Add(makeTestPodDetails("pod3", 5, "node2", map[string]string{"label": "value2"}))
+		store.bookmarkRevision <- 6
+		fc.Step(blockTimeout)
+
+		// Add an object to make sure the test would
+		// eventually fail instead of just waiting
+		// forever.
+		time.Sleep(10 * time.Second)
```

Comment:
I don't like sleeping in tests - can we do something smarter?

Reply:
Copied from https://github.com/kubernetes/kubernetes/blob/dccc75705d8584122279db5a6923f0e26db85565/staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache_test.go#L489-L515

Reply:
The fact that we've done something not the right way in the past doesn't mean we should continue making that.

We should figure out the right way and avoid sleeps.